pennylane/transforms/__init__.py

```diff
@@ -145,6 +145,7 @@
     ~transforms.make_tape
     ~transforms.create_expand_fn
     ~transforms.create_decomp_expand_fn
+    ~transforms.diagonalize_tape_measurements
```

Comment:
bad name?

Reply:
How about `diagonalize_for_measurements`, or just `diagonalize_measurements`?

Reply:
I like `diagonalize_measurements`.  I don't think we necessarily need `tape` in the name.